### ISSUE TITLE
Remove and rewire conflict 

### ISSUE THREAD

**Issue by cbler** [JKISoftware/JKI-State-Machine-Editor#57]
Seems to be a conflict in 2016 between the editor and the built-in right click menu item Remove and Rewire. Selecting this option removes and rewires, but then undoes the changes and results in no removal.

**Comment by jimkring** [comment]
Hi @cbler,

I tried to reproduce this in both 2015 and 2017 (sorry, I don't have 2016 handy at the moment).

Would you mind retesting with the [latest version](https://github.com/JKISoftware/JKI-State-Machine-Editor/releases/tag/2013.5.2.217) of the JKI State Machine Editor? Maybe we fixed the issue without realizing it.

And, if you still have the issue, could you give me some very specific steps to reproduce (and maybe a screenshot)?

Thanks!

**Comment by cbler** [comment]
Retested with version 2013.5.0.211 (LabVIEW 2016) and could easily reproduce and then noticed later version, 2013.5.3.218, is available. With same simple steps, was not able to reproduce. More complicated workflow, was able to reproduce. I'd call this behavior now just an annoyance, as it happens with much less frequency.

Steps
*New VI
*Add JKI SM 3.0.0.8 via pallete
*Duplicate "UI: Front Panel State"
*Save
*Run and press OK button on front panel to stop VI
*Start "Remove and Rewire" of "Add State(s) to Queue__JKI_lib_State_Machine.vi" on the separator no-op frames ("---------- Data ----------" for example)
*After the third or fourth "Remove and Rewire" of "Add State(s) to Queue__JKI_lib_State_Machine.vi" while working through the separator no-op frames, the VI goes away and then comes back

PS - love all the new features, thanks JKI team!

**Comment by jimkring** [comment]
That's certainly odd. I just tested in LabVIEW 2015 and after I select the "Duplicate Frame" option for the first time, I get another little pop-up menu showing *only* the "Remove and Rewire" right-click menu option (see screenshot).

![image](https://user-images.githubusercontent.com/381432/38149313-fd990c94-340e-11e8-9a11-3a0aa920cf0a.png)

This menu stays up and then if I click somewhere else on the screen that "Remove and Rewire" menu item goes away and then it does the Duplicate option (I can see the dialog asking me for the name of the new frame).

It seems to do this only once, after restarting LabVIEW and then it seems to work normally (the "Remove and Rewire" item doesn't pop up when it's not supposed to).

I didn't notice anything odd happen when I used the "Remove and Rewire" option in several separator frames as you mentioned.

**Comment by jimkring** [comment]
Any signs of this recently @cbler ? We were notified of a LabVIEW bug that was causing crashes and implemented a work-around.

See here: https://github.com/JKISoftware/JKI-State-Machine/issues/10

**Comment by cbler** [comment]
Yes, I've seen it occur very infrequently and noticed that if I repeat "Remove and Rewire" a second time it often will work. I consider this a very minor issue.

FWIW, since reporting this behavior, I've created my own Shortcut Menu Plugins (SMP) and noticed the same behavior with them occasionally as well. Also, I now recognized this behavior of a SMP (undoing its own steps) from debugging my own SMP (which sometimes had defects during development ;-). When I had a defect in my SMP, it would start doing what it was programmed to do, then encounter the error and undo the steps completed. I can speculate on what might be going on here; the SMP mechanism runs into an error at some point while iterating through all the SMPs after one is selected and that causes the undo to be done by the SMP mechanism.

Fortunately, I've not seen a crash of LabVIEW while using the JKISM Editor, as described in issue referenced.

BTW, this is still with LV2016 along with JKISM version v2018.0.5.41.

And lastly, I have really found the new features helpful - thanks again to the JKI team!

**Comment by cbler** [comment]
Follow-up (LV2016 along with JKISM version v2018.0.6.44):

Just got into this situation again, with the following workflow (tried to add as much detail as not sure what might be important):
- open VI with JKISM
- right click on diagram code to open JKISME
- create subvi of code in JKISM and start editing this new VI (JKISME disappears - very cool)
- run custom Shortcut Menu Plugin (SMP) on subvi (VI currently untitled and unsaved, hmmm) control and it works as expected
- run custom SMP on a different control and it works but then it undoes (perhaps due to some error in the SMP chain of plugins called) the changes and repeat SMP many times without success
-return to JKISM (JKISME reappears - extremely cool) and close JKISME window
-return to subvi and rerun SMP; success
So I do have another work around, it seems.
Cheers

**Comment by cbler** [comment]
Perhaps I have found a root cause, though it is a bit of speculation at this time.
This behaviour seems to occur (both to my SMPs and the JKISM Editor) if there is an untitled VI (which by definition is unsaved) present in my project - such as when blank VI is used for a new JKISM. It doesn't happen every time, but this strikes me as being the cause. Several times I have also seen this behaviour when I use Edit >> Create SubVI, leave the new VI untitled and run a SMP. It runs and then is undone - so the untitled VI might be causing an error not handled in my SMP.
